src/kOS.Safe/Compilation/Opcode.cs

```diff
@@ -1401,32 +1401,64 @@ public override string ToString()
         }
     }
     
+    /// <summary>
+    /// Returns from an OpcodeCall, popping a number of scope depths off
+    /// the stack as it does so.  It evals the topmost thing on the stack.
+    /// to remove any local variable references and replace them with their
+    /// current values, and then performs the equivalent of a popscope, then
+    /// jumps back to where the routine was called from.
+    /// It also checks to ensure that the argument stack contains the arg
+    /// bottom marker.  If it does not, that proves the number of parameters
+    /// consumed did not match the number of arguments passed and it throws
+    /// an exception (to avoid stack misalignment that would happen if it
+    /// tried to continue).
+    /// </summary>
```

Comment:
Half of what this is talking about was already true before, but as OpcodeReturn was getting complex now I decided it was best to document all of what it's doing.